cmd/cloud-run/signifysecretrotator/signifysecretrotator.py

```diff
@@ -0,0 +1,168 @@
+"""Simple PubSub handler to rotate signify certificates"""
+
+import datetime
+import os
+import base64
+import json
+import sys
+import traceback
+from typing import Any, Dict, List
+from flask import Flask, Response, request, make_response
+from cryptography import x509
+from cryptography.hazmat.primitives.asymmetric import rsa
+from cryptography.hazmat.primitives import serialization
+from cryptography.hazmat.primitives.serialization import Encoding, PrivateFormat
+from requests import HTTPError
+from secretmanager import client
+from pylogger.logger import Logger
+from signify.client import SignifyClient
+
+app = Flask(__name__)
+project_id: str = os.getenv("PROJECT_ID", "sap-kyma-prow")
+component_name: str = os.getenv("COMPONENT_NAME", "signify-certificate-rotator")
+application_name: str = os.getenv("APPLICATION_NAME", "secret-rotator")
+secret_rotate_message_type = os.getenv("SECRET_ROTATE_MESSAGE_TYPE", "signify")
+
+
+@app.route("/", methods=["POST"])
+def rotate_signify_secret() -> Response:
+    """HTTP webhook handler for rotating Signify secrets."""
+    logger = Logger(
+        component_name=component_name,
+        application_name=application_name,
+        request=request,
+    )
+
+    try:
+        sm_client = client.SecretManagerClient()
+
+        if project_id is None:
+            raise ValueError("Unknown project id")
+
+        pubsub_message = get_pubsub_message()
+
+        secret_rotate_msg = extract_message_data(pubsub_message)
+
+        # Pub/Sub topic handle multiple secret rotator components
+        # verify if we should handle that message
+        if secret_rotate_msg["labels"]["type"] != secret_rotate_message_type:
+            return prepare_error_response("Unsupported event type", logger)
+
+        secret_data = sm_client.get_secret(secret_rotate_msg["name"])
+
+        signify_client = SignifyClient(
+            token_url=secret_data["tokenURL"],
+            certificate_service_url=secret_data["certServiceURL"],
+            client_id=secret_data["clientID"],
+        )
+
+        old_cert_data = base64.b64decode(secret_data["certData"])
+        old_pk_data = base64.b64decode(secret_data["privateKeyData"])
+
+        if "password" in secret_data and secret_data["password"] != "":
+            old_pk_data = decrypt_private_key(
+                old_pk_data, secret_data["password"].encode()
+            )
+
+        new_private_key = rsa.generate_private_key(public_exponent=65537, key_size=4096)
```

Comment:
Magic Numbers

Reply:
It's not a magic numbers, both are described what they're

Reply:
Extracted 4096 to the top, added comment about public_exponent that is standarised and rsa module docs say it should be 65537 when it's not necessary to use 3.